spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/convert/EntityPropertyValueProvider.java

```diff
@@ -0,0 +1,70 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.datastore.core.convert;
+
+import com.google.cloud.datastore.Entity;
+
+import org.springframework.cloud.gcp.data.datastore.core.mapping.DatastoreDataException;
+import org.springframework.cloud.gcp.data.datastore.core.mapping.DatastorePersistentProperty;
+import org.springframework.data.mapping.model.PropertyValueProvider;
+import org.springframework.util.ClassUtils;
+
+/**
+ * A {@link PropertyValueProvider} for Datastore entities
+ *
+ * @author Dmitry Solomakha
+ *
+ * @since 1.1
+ */
+public class EntityPropertyValueProvider implements PropertyValueProvider<DatastorePersistentProperty> {
```

Comment:
Let's have a test for this class as well, even though it's used by the converter.